common/src/main/java/tachyon/client/BlockMasterClient.java

```diff
@@ -0,0 +1,242 @@
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Return the BlockInfo for a block id
+   *
+   * @param blockId the block id to get the BlockInfo for
+   * @return the BlockInfo
+   * @throws IOException
+   */
+  public synchronized BlockInfo getBlockInfo(long blockId) throws IOException {
+    while (!mIsClosed) {
+      connect();
+      try {
+        return mClient.getBlockInfo(blockId);
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Get the total Tachyon capacity in bytes, on all the tiers of all the workers.
+   *
+   * @return total capacity in bytes
+   * @throws IOException
+   */
+  public synchronized long getCapacityBytes() throws IOException {
+    while (!mIsClosed) {
+      connect();
+      try {
+        return mClient.getCapacityBytes();
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    return -1;
+  }
+
+  /**
+   * Get the total amount of used space in bytes, on all the tiers of all the workers.
+   *
+   * @return amount of used space in bytes
+   * @throws IOException
+   */
+  public synchronized long getUsedBytes() throws IOException {
+    while (!mIsClosed) {
+      connect();
+      try {
+        return mClient.getUsedBytes();
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    return -1;
+  }
+
+  // TODO: split out the following worker specific interactions to a separate block master client
+  // for the worker.
+
+  /**
+   * Commits a block on a worker.
+   *
+   * @param workerId the worker id committing the block
+   * @param usedBytesOnTier the amount of used bytes on the tier the block is committing to
+   * @param tier the tier the block is being committed to
+   * @param blockId the block id being committed
+   * @param length the length of the block being committed
+   * @throws IOException
+   */
+  public synchronized void workerCommitBlock(long workerId, long usedBytesOnTier, int tier, long
+      blockId, long length) throws IOException {
+    while (!mIsClosed) {
+      connect();
+      try {
+        mClient.workerCommitBlock(workerId, usedBytesOnTier, tier, blockId, length);
+        return;
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+  }
+
+  /**
+   * Returns a worker id for a workers net address.
+   *
+   * @param address the net address to get a worker id for
+   * @return a worker id
+   * @throws IOException
+   */
+  public synchronized long workerGetId(NetAddress address) throws IOException {
+    while (!mIsClosed) {
+      connect();
+      try {
+        return mClient.workerGetWorkerId(address);
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
```

Comment:
is it possible to have this loop an infinite loop, and logging becomes too much to kill the performance?